app/controllers/clinicfinders_controller.rb

```diff
@@ -1,14 +1,17 @@
 # Controller for use with clinic_finder gem; locates nearest clinic
 # based on certain info.
 class ClinicfindersController < ApplicationController
-  def index; end
-
   def search
-    @abortron = Abortron::ClinicFinder.new("#{Rails.root}/clinics.yml")
-    @nearest = @abortron.locate_nearest_clinic patient_zipcode: params[:zip].to_i.to_s,
-                                               gestational_age: params[:gestation].to_i
-    @cheapest = @abortron.locate_cheapest_clinic(gestational_age: params[:gestation].to_i)
+    return head :bad_request if params[:zip].blank?
+
+    @abortron = ClinicFinder::Locator.new(
```

Comment:
Is abortron the right name for this?

Reply:
Probably not, haha. It also doesn't need to be an instance variable, so I'll rename it to something like `finder` and localize it.